src/test/java/teammates/test/cases/BaseTestCaseWithDatastoreAccess.java

```diff
@@ -187,6 +187,8 @@ private void equalizeIrrelevantData(AccountAttributes expected, AccountAttribute
                 expected.studentProfile.googleId = actual.googleId;
             }
             expected.studentProfile.modifiedDate = actual.studentProfile.modifiedDate;
+            // data is expected to be sanitized
+            expected.sanitizeForSaving();
```

Comment:
We have full control of the `expected` data, right? Does that not mean that we can self-sanitize (I wonder if such term exists?) the data in `expected`?